src/components/setup/setup-card.tsx

```diff
@@ -0,0 +1,198 @@
+'use client';
+
+import { Setup } from '@/actions/setup/get-setup';
+import { toggleLikeSetup } from '@/actions/setup/like-setup';
+import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
+import { Button } from '@/components/ui/button';
+import { Card, CardContent, CardFooter, CardHeader } from '@/components/ui/card';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Session, User } from '@prisma/client';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { Edit, Heart, MoreVertical, Trash } from 'lucide-react';
+import { useOptimistic, useState } from 'react';
+
+interface SetupCardProps {
+  setup: Setup;
+  session: (Session & { user: User }) | null;
+  onDelete: () => void;
+  onEdit: () => void;
+  onRequireAuth?: () => void;
+}
+
+export function SetupCard({ setup, session, onDelete, onEdit, onRequireAuth }: SetupCardProps) {
+  const [imgError, setImgError] = useState(false);
+  const [isLiking, setIsLiking] = useState(false);
+  const [optimisticLikes,] = useOptimistic(
+    setup.likes,
+    (state, userId: string) => {
+      const isLiked = state.some((like) => like.userId === userId);
+      return isLiked ? state.filter((like) => like.userId !== userId) : [...state, { userId }];
+    },
+  );
+  const queryClient = useQueryClient();
```

Comment:
Acá podriamos agregar algunas lineas en blanco para mejorar la legibilidad
```suggestion
  const [imgError, setImgError] = useState(false);
  const [isLiking, setIsLiking] = useState(false);
  
  const [optimisticLikes,] = useOptimistic(
    setup.likes,
    (state, userId: string) => {
      const isLiked = state.some((like) => like.userId === userId);
      return isLiked ? state.filter((like) => like.userId !== userId) : [...state, { userId }];
    },
  );
  
  const queryClient = useQueryClient();
```